dynamically_sized_list.c

```diff
@@ -0,0 +1,243 @@
+#include "stdlib.h"
+#include "string.h"
+#include "stdio.h"
+#include "dynamically_sized_list.h"
+
+void readStringFromConsole(char** str_)
+{
+	/*	
+		this function allocate dynamic memory for sentence in stdin buffer (until '\n' is reached)
+		and assign this string to str_
+	*/
+	if (!str_) return;
+	int reallocSize = REALLOC_SIZE, // value of memory to be added every reallocation
+		offset = 0; //current position
+	char* str = NULL;
+	do {
+		str = (char*)malloc(reallocSize * sizeof(char));

```

Comment:
The 'str' pointer was assigned values twice without releasing the memory. A memory leak is possible.